### ISSUE TITLE
Mapping search results

### ISSUE THREAD

**Issue by sdfordham** [AnthonyBloomer/daftlistings#63]
On daft.ie you can view a map of properties in your search result, but it doesn't work properly and never has because it shows properties that have already been let/sold rendering it useless. So it could be nice to add this functionality to this module.

I have played with [folium](https://pypi.org/project/folium/) to do this. Here is some example code (ignore the function definition it will go when the `fetchall` argument is added to `.search()`
````python
import folium
from daftlistings import Daft, RentType

#---------------------------------
def get_all_listings(daft):
    listings = daft.search()
    tot_listings = listings
    offset = 0
    
    while len(listings) == 20:
        offset = offset + 20
        daft.set_offset(offset)
        listings = daft.search()
        tot_listings = tot_listings + listings
    return tot_listings
#---------------------------------

daft = Daft()

daft.set_county("Dublin City")
daft.set_listing_type(RentType.APARTMENTS)
daft.set_min_price(1000)
daft.set_max_price(1500)

listings = get_all_listings(daft)

# Center the map on first listing
m = folium.Map(location=[float(listings[0].latitude),
                         float(listings[0].longitude)],
               zoom_start=13)

# Add a marker for each listing.
# On clicking the marker show price and a weblink to the listing
for l in listings:
    folium.Marker([float(l.latitude),
                   float(l.longitude)],
                  popup=l.price + ' <a href="' + l.daft_link + '">(link)</a>').add_to(m)
m
````
One issue is with folium is it seems to be slow generating the map.

**Comment by AnthonyBloomer** [comment]
@sdfordham, that is so cool. Great visualisation using folium. I have released [v1.7.4](https://pypi.org/project/daftlistings/)  which includes the `fetch_all` parameter in `search`. I've made the default value to be True so it will fetch all listings by default. Here's an example based on the code you've added:

``` python
import folium
from daftlistings import Daft, RentType

daft = Daft()

daft.set_county("Dublin City")
daft.set_listing_type(RentType.APARTMENTS)
daft.set_min_price(1500)
daft.set_max_price(1500)

listings = daft.search()

m = folium.Map(location=[float(listings[0].latitude),
                         float(listings[0].longitude)],
               zoom_start=13)

m

# Add a marker for each listing.
# On clicking the marker show price and a weblink to the listing
for l in listings:
    folium.Marker([float(l.latitude),
                   float(l.longitude)],
                  popup=l.price + ' <a href="' + l.daft_link + '">(link)</a>').add_to(m)

m
```

In terms of integrating this with this module, one way would be to add a `generate_map` function in the `Daft` class, for example:

``` python
daft.set_county("Dublin City")
daft.set_listing_type(RentType.APARTMENTS)
daft.set_min_price(1500)
daft.set_max_price(1500)

listings = daft.search(fetch_all=True)
m = daft.generate_map(listings)
```

That function could simply be a wrapper around the code you've wrote to generate the map or it could be altered further to allow for additional functionality integrated. I'd like to hear your thoughts.

**Comment by sdfordham** [comment]
It seems like a good starting point. My only concern is folium seems a little slow. But maybe it has to be, I guess generating a map has a fair bit of overhead involved.
